### ISSUE TITLE
Finetune code

### ISSUE THREAD

**Issue by msamwelmollel** [kyutai-labs/hibiki#15]
How can I finetune HIBIKI for my other dataset

**Comment by LaurentMazare** [comment]
Hibiki uses the moshi codebase and the same architecture, so you can try tweaking the [kyutai-labs/moshi-finetune](https://github.com/kyutai-labs/moshi-finetune) codebase to adapt it to your datasets.

**Comment by msamwelmollel** [comment]
Please correct me if I'm wrong, and my dataset is voice-to-voice in Hausa and English. However, the Moshi dataset seems quite different. Could someone help clarify this or at least provide a minimal example for fine-tuning Hibiki?

> > Hibiki uses the moshi codebase and the same architecture, so you can try tweaking the [kyutai-labs/moshi-finetune](https://github.com/kyutai-labs/moshi-finetune) codebase to adapt it to your datasets.
> 
> could you provide an example of hibiki data format

**Comment by msamwelmollel** [comment]
Hello Team Moshi and Hibiki @LaurentMazare @adefossez 

Below is a **cleaned-up summary** of my current situation and what I need help with:


### **How I'm Preparing My Dataset and Pipeline**

1. **Single JSONL File** (`data.jsonl`):  
   ```json
   {"path": "/path/Speech-To-Speech-Translation/data/eng_audio/STEREO_common_voice_sw_30623601.mp3.wav", "duration": 19.076}
   ```
   - This points to **one stereo wave** file where **channel 1** is Swahili (source), and **channel 2** is English (target).

2. **Alignments for Target (English) Audio**:  
   ```json
   {
     "alignments": [
       ["He", [4.2, 5.66], "SPEAKER_MAIN"],
       ["loved", [5.66, 5.9], "SPEAKER_MAIN"],
       ...
       ["kind.", [8.48, 8.78], "SPEAKER_MAIN"]
     ]
   }
   ```
   - These timestamps describe the target (English) words in channel 2.  
   - My goal is for the **target audio** to start only **after** the source has finished talking (like consecutive translation).

3. **Interleaver Config**:  
   - I changed `keep_main_only=True` → `False`, hoping it would include both channels or avoid dropping certain alignments.

4. **Loss Calculation**:  
   - I'm still using `mb_loss = text_loss + audio_loss`, expecting that the text tokens might help improve translation quality.



### **Train.py and Config**

- **Train Script**: I didn’t rewrite much beyond switching `keep_main_only=False`.  
- **YAML Config** snippet:
  ```yaml
  data:
    train_data: '/path/data/data.jsonl'
    shuffle: true

  moshi_paths:
    hf_repo_id: "kyutai/hibiki-1b-pytorch-bf16"

  full_finetuning: false
  lora:
    enable: true
    rank: 512
    scaling: 2.0

  first_codebook_weight_multiplier: 100.0
  text_padding_weight: 0.5

  duration_sec: 20
  batch_size: 4
  max_steps: 250
  gradient_checkpointing: true

  optim:
    lr: 2.e-4
    weight_decay: 0.1
    pct_start: 0.05

  run_dir: "/path/Speech-To-Speech-Translation/test"
  ```
- I’m using partial finetuning with LoRA, a fairly large LR (2e-4), a short max sequence length (20s), etc.

---

### **Core Issue**

**After training**, my model doesn’t produce any audible output at inference time (it “doesn’t speak”). I’m trying to figure out why. The stereo wave is set up so:
- **Channel 1 (source)**: Swahili, no text alignment.
- **Channel 2 (target)**: English, with `SPEAKER_MAIN` alignments.

I assume the source chunk ends, and then the target chunk should come in. But I’m not getting any voiced output.

1. I might be missing something in how I feed the stereo wave → separate channels.  
2. Or maybe the `keep_main_only=False` isn’t enough to keep the relevant alignments.  
3. Possibly the large LR or short sequence context is an issue.  
4. Or maybe my pipeline for generating (inference) is incomplete.

**Any guidance** on making the model produce the target speech is appreciated!
